src/textual/css/styles.py

```diff
@@ -581,7 +584,7 @@ def partial_rich_style(self) -> Style:
 @dataclass
 class Styles(StylesBase):
     node: DOMNode | None = None
-    _rules: RulesMap = field(default_factory=dict)
+    _rules: RulesMap = field(default_factory=RulesMap.__call__)
```

Comment:
Why is the `__call__` necessary here?

Reply:
mypy sees `RulesMap` as `Type[RulesMap]` instead of `() -> RulesMap`, which is what we want.
The error with `RulesMap` only:
```
src/textual/css/styles.py:587: error: Argument "default_factory" to "field" has incompatible type "Type[RulesMap]"; expected "Callable[[], RulesMap]"
```

I can replace `RulesMap.__call__` with `cast(Callable[[], RulesMap], RulesMap)` but I don't like `cast` very much and that is why I went with `RulesMap.__call__`. I can also add an explicit comment like `# mypy isn't happy with just RulesMap`.

Reply:
I guess Mypy doesn't understand that types are also callables.

would `default_factory = lambda: RulesMap()` be preferable here?

Reply:
It satisfies mypy and I think it looks better, so I replaced it.